### PR TITLE
Refactor migration token validation and match entire token on endpoints

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

This PR makes changes to how the migration token is used for user migration scenarios:

- Migration tokens will now be compared first char-to-char with what was sent, then by decoding and checking for a `jti` attribute. 
- No connection updates are attempted when turning the endpoints on/off (this was not working anyways as the connection ID is not stored)
- New tokens generated when the service is turned on for the first time are not JWTs
- Admin documentation clarified

### Testing

* [x] This change adds unit test coverage
* [x] This change has been tested on the latest version of WP

### Checklist

* [x] All existing and new tests complete without errors
* [x] All code quality tools/guidelines in the [Contribution guide](CONTRIBUTION.md) have been run/followed
* [x] All active GitHub CI checks have passed

### CODE REVIEW

File: lib/WP_Auth0_Routes.php [hunk 1]
```diff
@@ -158,7 +158,7 @@ protected function getAuthorizationHeader() {
 	protected function migration_ws_login() {
 
 		// Migration web service is not turned on.
-		if ( $this->a0_options->get( 'migration_ws' ) == 0 ) {
+		if ( ! $this->a0_options->get( 'migration_ws' ) ) {
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: More accurate checking for this setting.

File: lib/WP_Auth0_Routes.php [hunk 2]
```diff
@@ -173,18 +173,13 @@ protected function migration_ws_login() {
 		$authorization = $this->getAuthorizationHeader();
 		$authorization = trim( str_replace( 'Bearer ', '', $authorization ) );
 
-		$secret   = $this->a0_options->get_client_secret_as_key( true );
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: Moved to `$this->valid_token`

File: lib/WP_Auth0_Routes.php [hunk 3]
```diff
@@ -237,21 +232,15 @@ protected function migration_ws_get_user() {
 
 		$authorization = $this->getAuthorizationHeader();
 		$authorization = trim( str_replace( 'Bearer ', '', $authorization ) );
-
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: Moved to `$this->valid_token`

File: lib/WP_Auth0_Routes.php [hunk 4]
```diff
@@ -223,7 +218,7 @@ protected function migration_ws_login() {
 	protected function migration_ws_get_user() {
 
 		// Migration web service is not turned on.
-		if ( $this->a0_options->get( 'migration_ws' ) == 0 ) {
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: More accurate checking for this setting.

File: lib/admin/WP_Auth0_Admin_Advanced.php [hunk 5]
```diff
@@ -705,63 +706,51 @@ public function basic_validation( $old_options, $input ) {
 		return $input;
 	}
 
-	public function migration_ws_validation( $old_options, $input ) {
```
Comment (joshcanhelp) [inline lib/admin/WP_Auth0_Admin_Advanced.php]: Rewritten, probably best to review the final file.

File: lib/WP_Auth0_Routes.php [hunk 6]
```diff
@@ -237,21 +232,15 @@ protected function migration_ws_get_user() {
 
 		$authorization = $this->getAuthorizationHeader();
 		$authorization = trim( str_replace( 'Bearer ', '', $authorization ) );
-
-		$secret   = $this->a0_options->get_client_secret_as_key( true );
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: Moved to `$this->valid_token`

File: lib/WP_Auth0_Routes.php [hunk 7]
```diff
@@ -317,6 +306,32 @@ private function error_return_array( $code ) {
...
+	}
+
+	/**
+	 * Check if a token or token JTI is the same as what is stored.
+	 *
+	 * @param string $authorization - Incoming migration token;
+	 *
+	 * @return bool
+	 */
+	private function valid_token( $authorization ) {
```
Comment (cocojoe) [inline lib/WP_Auth0_Routes.php]: It possible to end up in some kind of RS256 situation? or it can only ever be HS256?

Comment (joshcanhelp) [reply lib/WP_Auth0_Routes.php]: No. Original token was always created with HS256 but this new version compares char to char so the algorithm doesn't matter any longer.

**Pull request merged**